### PR TITLE
ADEN Refactor SignupTests

### PR DESCRIPTION
**Pull request by drets**
@tomasznapieralski what do you think this refactoring makes the tests more readable?
If Java would support the named arguments we could do more fancy stuff like:
```java
 public void MercurySignupTest_004_signupErrorBadPassword() {
   signUp(password: INCORRECT_PASSWORD)
       .verifyPasswordError();
 }

public void MercurySignupTest_003_signupErrorUsernameTaken() {
  signUp(userName: TAKEN_USERNAME)
      .verifyUsernameTakenError();
}
```

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/SignupPageObject.java [hunk 1]
```diff
@@ -50,25 +47,29 @@ public SignupPageObject(WebDriver driver) {
...
+  private SignupPageObject typePassword(String password) {
     wait.forElementVisible(signupPassword);
     signupPassword.sendKeys(password);
     return this;
   }
 
-  public SignupPageObject typeBirthdate(String month, String day, String year) {
+  private SignupPageObject typeBirthdate(String month, String day, String year) {
     wait.forElementVisible(signupBirthdate);
     scrollAndClick((signupBirthdate));
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/SignupPageObject.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Remove those useless parentheses.](https://github.com/Wikia/selenium-tests/blob/59e0235c43befc8333b8e0e5f2df000541131116/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/SignupPageObject.java#L77) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/SignupPageObject.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import com.wikia.webdriver.common.core.Assertion;
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/SignupPageObject.java]: I would recommend using builder, so you can pass a user object like this: 
```
signUp(CreateUser().withPassword(WRONG_PASS).create())
                      .verifyEmailInUseError();

signUp(CreateUser(),withName(NAME_TAKEN).create())
                     .verifyEmailInUseError();
```

**Comment by ludwikkazmierczak** [comment]
@drets functionality you are talking about is easy to implement with builder pattern, It would look like this:
```
signUp(CreateUser().withPassword(WRONG_PASS).create())
                      .verifyEmailInUseError();

signUp(CreateUser(),withName(NAME_TAKEN).create())
                     .verifyEmailInUseError();
```

**Comment by tomasznapieralski** [comment]
It looks nice, thanks @drets !

**Comment by drets** [comment]
@ludwikkazmierczak thanks! 
Using builder pattern makes perfect sense to me :wink:
However, even this version of tests look much more readable and the PR is just an addition on this comment: https://github.com/Wikia/selenium-tests/pull/804#discussion_r36891152
So, I would consider a refactor to use builder pattern in the future PR :smile_cat:

File: src/test/java/com/wikia/webdriver/common/users/CreateUser.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import org.joda.time.DateTime;
+
+public class CreateUser {
+  private String userName = "User" + DateTime.now().getMillis();
+  private String password = "Pass" + DateTime.now().getMillis();
+  private String email = "qaart001+" + DateTime.now().getMillis() + "@gmail.com";
+  private DateTime birthday = new DateTime(1952, 12, 12, 12, 0, 0);
+
+  public CreateUser() {}
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/users/CreateUser.java]: MAJOR Move this closing curly brace to the next line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by ludwikkazmierczak** [comment]
Added builder to this one

**Comment by drets** [comment]
LGTM :gem:

**Comment by drets** [comment]
LGTM :gem:

**Comment by tomasznapieralski** [comment]
LGTM, can we merge it or it is still onhold?

**Comment by drets** [comment]
@tomasznapieralski I think so :wink:

**Comment by tomasznapieralski** [comment]
All tests passed locally and on Jenkins

**Pull request merged**